### DISCUSSION TITLE
When does Karpenter bind the pod to the node？

### DISCUSSION THREAD

**Discussion by bibibox** [kubernetes-sigs/karpenter#859] [Q&A] [answered]
i find this introduction about it:
`Scheduling enforcement: After determining launch capacity and scheduling constraints, Karpenter optimistically creates the Node object and binds the pod to it immediately. In contrast, Cluster Autoscaler does not bind pods to the nodes it creates. Instead, it relies on the kube-scheduler to make the same scheduling decision after the node has come online.`

but i cannot find the bind action from source code, and i find a doc which said the karpenter has removed the create node action yet: [https://github.com/aws/karpenter-provider-aws/blob/main/designs/node-ownership.md](url), so when does Karpenter bind the pod to the node now? Would it be nice to point out where this is in the source code, Thanks!

**Comment by bibibox** [comment]
can anyone help me? Thanks!

**Comment by tzneal** [comment, marked as answer]
Karpenter no longer binds pods to the node. It was removed many versions ago when the Karpenter scheduler got to the point that it was no longer necessary.
